src/a11y-dialog.js

```diff
@@ -1,283 +1,237 @@
-  this._openers.forEach(
-    function (opener) {
-      opener.addEventListener('click', this._show)
-    }.bind(this)
-  )
-
-  // Keep a collection of dialog closers, each of which will be bound a click
-  // event listener to close the dialog
-  this._closers = $$('[data-a11y-dialog-hide]', this.$el).concat(
-    $$('[data-a11y-dialog-hide="' + this._id + '"]')
-  )
-  this._closers.forEach(
-    function (closer) {
-      closer.addEventListener('click', this._hide)
-    }.bind(this)
-  )
-
-  // Execute all callbacks registered for the `create` event
-  this._fire('create')
-}
 
-/**
- * Show the dialog element, trap the current focus within it, listen for some
- * specific key presses and fire all registered callbacks for `show` event
- *
- * @param {CustomEvent} event - An optional `CustomEvent`, passed to `show`
- *                              event listeners
- * @return {A11yDialog}
- */
-A11yDialog.prototype.show = function (event) {
-  // If the dialog is already open, abort
-  if (this.shown) {
-    return this
-  }
+/** @type A11yDialogType */
+export default class A11yDialog {
+  /** @param {element} HTMLElement */
+  constructor(element) {
+    this.$el = element
+    this.id = this.$el.getAttribute('data-a11y-dialog') || this.$el.id
 
-  // Keep a reference to the currently focused element to be able to restore
-  // it later
-  this._previouslyFocused = document.activeElement
-  this.$el.removeAttribute('aria-hidden')
-  this.shown = true
+    this.$el.setAttribute('aria-hidden', 'true')
+    this.$el.setAttribute('aria-modal', 'true')
+    this.$el.setAttribute('tabindex', '-1')
 
-  // Set the focus to the dialog element
-  moveFocusToDialog(this.$el)
+    if (!this.$el.hasAttribute('role')) {
+      this.$el.setAttribute('role', 'dialog')
+    }
 
-  // Bind a focus event listener to the body element to make sure the focus
-  // stays trapped inside the dialog while open, and start listening for some
-  // specific key presses (TAB and ESC)
-  document.body.addEventListener('focus', this._maintainFocus, true)
-  document.addEventListener('keydown', this._bindKeypress)
+    // Keep a collection of dialog openers, each of which will be bound a click
+    // event listener to open the dialog
+    this.openers = $$(`[data-a11y-dialog-show="${this.id}"]`)
+    this.openers.forEach(opener => {
+      opener.addEventListener('click', this.show)
+    })
+
+    // Keep a collection of dialog closers, each of which will be bound a click
+    // event listener to close the dialog
+    this.closers = $$(
+      `[data-a11y-dialog="${this.id}"] [data-a11y-dialog-hide], #${this.id} [data-a11y-dialog-hide], [data-a11y-dialog-hide="${this.id}"]`
+    )
+    this.closers.forEach(closer => {
+      closer.addEventListener('click', this.hide)
+    })
 
-  // Execute all callbacks registered for the `show` event
-  this._fire('show', event)
+    this.listeners = {}
 
-  return this
-}
+    this.previouslyFocused = null
```

Comment:
I kind of liked to have a leading underscore for properties and methods that are not part of the public API. What do you think?

Reply:
I like it when we have no way in a language to indicate privacy to the dev, but we can use the keyword “private” in TypeScript, so I think I would prefer not to have leading underscores since we’re about to migrate.

Reply:
We’ll need to clarify the name change in the changelog though, since some people might be relying on these “hidden” properties.